sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala

```diff
@@ -413,38 +413,36 @@ case class DescribeTableCommand(table: TableIdentifier, isExtended: Boolean, isF
     } else {
       val metadata = catalog.getTableMetadata(table)
 
+      if (DDLUtils.isDatasourceTable(metadata)) {
+        DDLUtils.getSchemaFromTableProperties(metadata) match {
+          case Some(userSpecifiedSchema) => describeSchema(userSpecifiedSchema, result)
+          case None => describeSchema(catalog.lookupRelation(table).schema, result)
+        }
+      } else {
+        describeSchema(metadata.schema, result)
+      }
```

Comment:
How about we try to put these into describeSchema? Of, maybe we can add a `describeSchema(tableName, result)`? Seems it is weird that `describeExtended` and `describeFormatted` do not contain the code for describing the schema.